CMakeLists.txt

```diff
@@ -19,46 +19,29 @@ cmake_minimum_required(VERSION 3.12)
 
 project(vulkan_samples)
 
-# create output folder
-file(MAKE_DIRECTORY output)
-
 # Add path for local cmake scripts
 list(APPEND CMAKE_MODULE_PATH
     ${CMAKE_CURRENT_SOURCE_DIR}/bldsys/cmake
     ${CMAKE_CURRENT_SOURCE_DIR}/bldsys/cmake/module)
 
-# Precompiled header macro. Parameters are source file list and filename for pch cpp file.
-macro(vulkan_samples_pch SRCS PCHCPP)
-  if(MSVC AND CMAKE_GENERATOR MATCHES "^Visual Studio")
-    set(PCH_NAME "$(IntDir)\\pch.pch")
-    # make source files use/depend on PCH_NAME
-    set_source_files_properties(${${SRCS}} PROPERTIES COMPILE_FLAGS "/Yupch.h /FIpch.h /Fp${PCH_NAME} /Zm300" OBJECT_DEPENDS "${PCH_NAME}")
-    # make PCHCPP file compile and generate PCH_NAME
-    set_source_files_properties(${PCHCPP} PROPERTIES COMPILE_FLAGS "/Ycpch.h /Fp${PCH_NAME} /Zm300" OBJECT_OUTPUTS "${PCH_NAME}")
-    list(APPEND ${SRCS} "${PCHCPP}")
-  endif()
-endmacro(vulkan_samples_pch)
-
 include(utils)
-include(global_options)
+include(global_config)
 include(sample_helper)
 include(check_atomic)
+include(test_helpers)
+
+# options
+set(VKB_WARNINGS_AS_ERRORS ON CACHE BOOL "Enable Warnings as Errors")
+set(VKB_VALIDATION_LAYERS OFF CACHE BOOL "Enable validation layers for every application.")
+set(VKB_VALIDATION_LAYERS_GPU_ASSISTED OFF CACHE BOOL "Enable GPU assisted validation layers for every application.")
+set(VKB_VULKAN_DEBUG ON CACHE BOOL "Enable VK_EXT_debug_utils or VK_EXT_debug_marker if supported.")
+set(VKB_BUILD_SAMPLES ON CACHE BOOL "Enable generation and building of Vulkan best practice samples.")
+set(VKB_BUILD_TESTS OFF CACHE BOOL "Enable generation and building of Vulkan best practice tests.")
+set(VKB_WSI_SELECTION "XCB" CACHE STRING "Select WSI target (XCB, XLIB, WAYLAND, D2D)")
 
 # Add third party libraries
 add_subdirectory(third_party)
 
-# Add vulkan framework
-add_subdirectory(framework)
-
-if(VKB_BUILD_TESTS)
-    # Add vulkan tests
-    add_subdirectory(tests)
-endif()
-
-if(VKB_BUILD_SAMPLES)
-    # Add vulkan samples
-    add_subdirectory(samples)
-endif()
+vkb__enable_testing()
 
-# Add vulkan app (runs all samples)
-add_subdirectory(app)
+add_subdirectory(components)
```

Comment:
I don't remember where, but I've seen warnings in the past for no new line at end of documents.

Reply:
Yeah, it's really not a big issue, just seen the warnings and now am a diligent new line at end of file devotee..